### ISSUE TITLE
energy function can output NaN if compiled

### ISSUE THREAD

**Issue by ThoreWietzke** [AaltoML/BayesNewton#27]
I'm using the MarkovVariationalGP model to compute the energy function. When I compile the energy function, the returned values are not the same, when I use the non compiled function. Sometimes, I get a NaN as output.

My minimal example is the following:
```python
x = np.arange(0., 4., 0.05)
y = x**2
var_f = 1.
var_y = 1.
lengthscale_f = 1.

kern = bayesnewton.kernels.Matern32(variance=var_f, lengthscale=lengthscale_f)
lik = bayesnewton.likelihoods.Gaussian(variance=var_y)
model = bayesnewton.models.MarkovVariationalGP(kernel=kern, likelihood=lik, X=x, Y=y)

energy = objax.GradValues(model.energy, model.vars())

energy_Jit = objax.Jit(energy, model.vars())
dE, E = energy()
dE_comp, E_comp = energy_Jit()

print(f"Energy={E}, Gradient of Energy={dE}")
print(f"EnergyComp={E_comp}, Gradient of EnergyComp={dE_comp}")
```

I digged further and the NaN is the result of `gaussian_expected_log_lik` in `utils`. The masks used here are not the same in the compiled and uncompiled version. The used mask here is `mask_pseudo_y` from the BaseModel. Especially `mask` and `maskv` can have different values which result in `noise=0`. This produces a NaN in`mvn_logpdf`.

I've included the function here for easier access:
```python
def gaussian_expected_log_lik(Y, q_mu, q_covar, noise, mask=None):
    """
    :param Y: N x 1
    :param q_mu: N x 1
    :param q_covar: N x N
    :param noise: N x N
    :param mask: N x 1
    :return:
        E[log 𝓝(yₙ|fₙ,σ²)] = ∫ log 𝓝(yₙ|fₙ,σ²) 𝓝(fₙ|mₙ,vₙ) dfₙ
    """
    if mask is not None:
        # build a mask for computing the log likelihood of a partially observed multivariate Gaussian
        maskv = mask.reshape(-1, 1)
        jax.debug.print("mask={mask}, maskv={maskv}", mask=mask, maskv=maskv)
        q_mu = np.where(maskv, Y, q_mu)
        noise = np.where(maskv + maskv.T, 0., noise)  # ensure masked entries are independent
        noise = np.where(np.diag(mask), INV2PI, noise)  # ensure masked entries return log like of 0
        q_covar = np.where(maskv + maskv.T, 0., q_covar)  # ensure masked entries are independent
        q_covar = np.where(np.diag(mask), 1e-20, q_covar)  # ensure masked entries return trace term of 0

    ml = mvn_logpdf(Y, q_mu, noise)
    trace_term = -0.5 * np.trace(solve(noise, q_covar))
    return ml + trace_term
```

In my opinion, the compiled and uncompiled versions should output the same value.

**Comment by ThoreWietzke** [comment]
Found out it's a problem from the current Windows build for jax. So no action has to be taken here ...
